File: sdkwork-ai-sdk-csharp/Api/ChatApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class ChatApi
    {
        private readonly HttpClient _client;

        public ChatApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Get chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> GetCompletionAsync(string completion_id)
        {
            return await _client.GetAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/chat/completions/{completion_id}"));
        }

        /// <summary>
        /// Update chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> CreateUpdateCompletionAsync(string completion_id, UpdateCompletionPostRequest body)
        {
            return await _client.PostAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/chat/completions/{completion_id}"), body);
        }

        /// <summary>
        /// Delete chat completion
        /// </summary>
        public async Task<ChatCompletionDeleteResponse?> DeleteCompletionAsync(string completion_id)
        {
            return await _client.DeleteAsync<ChatCompletionDeleteResponse>(ApiPaths.AiPath($"/chat/completions/{completion_id}"));
        }

        /// <summary>
        /// Update chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> PatchUpdateCompletionAsync(string completion_id, UpdateCompletionPatchRequest body)
        {
            return await _client.PatchAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/v1/chat/completions/{completion_id}"), body);
        }

        /// <summary>
        /// Update chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> PatchUpdateCompletionChatAsync(string completion_id, PatchUpdateCompletionRequest body)
        {
            return await _client.PatchAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/chat/completions/{completion_id}"), body);
        }

        /// <summary>
        /// Get chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> GetChatCompletionAsync(string completion_id)
        {
            return await _client.GetAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/management/chat/completions/{completion_id}"));
        }

        /// <summary>
        /// Update chat completion
        /// </summary>
        public async Task<ChatCompletionResponse?> UpdateChatCompletionAsync(string completion_id, UpdateChatCompletionRequest body)
        {
            return await _client.PostAsync<ChatCompletionResponse>(ApiPaths.AiPath($"/management/chat/completions/{completion_id}"), body);
        }

        /// <summary>
        /// Delete chat completion
        /// </summary>
        public async Task<ChatCompletionDeleteResponse?> DeleteChatCompletionAsync(string completion_id)
        {
            return await _client.DeleteAsync<ChatCompletionDeleteResponse>(ApiPaths.AiPath($"/management/chat/completions/{completion_id}"));
        }

        /// <summary>
        /// List chat completions
        /// </summary>
        public async Task<ChatCompletionList?> ListCompletionsAsync(Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<ChatCompletionList>(ApiPaths.AiPath("/chat/completions"), query);
        }

        /// <summary>
        /// Create chat completion
        /// </summary>
        public async Task<CreateChatCompletionResponse?> CreateChatCompletionAsync(ChatCompletionRequest body)
        {
            return await _client.PostAsync<CreateChatCompletionResponse>(ApiPaths.AiPath("/chat/completions"), body);
        }

        /// <summary>
        /// Get chat messages
        /// </summary>
        public async Task<ChatMessageList?> GetChatMessagesAsync(string completion_id, Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<ChatMessageList>(ApiPaths.AiPath($"/management/chat/completions/{completion_id}/messages"), query);
        }

        /// <summary>
        /// List chat completions
        /// </summary>
        public async Task<ChatCompletionList?> ListChatCompletionsAsync(Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<ChatCompletionList>(ApiPaths.AiPath("/management/chat/completions"), query);
        }

        /// <summary>
        /// Get chat completion messages
        /// </summary>
        public async Task<ChatMessageList?> GetMessagesAsync(string completion_id, Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<ChatMessageList>(ApiPaths.AiPath($"/chat/completions/{completion_id}/messages"), query);
        }
    }
}

File: sdkwork-ai-sdk-swift/Sources/API/ChatApi.swift
import Foundation

public class ChatApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Get chat completion
    public func getCompletion(completion_id: String) async throws -> ChatCompletionResponse? {
        let response = try await client.get(ApiPaths.aiPath("/chat/completions/\(completion_id)"))
        return response as? ChatCompletionResponse
    }

    /// Update chat completion
    public func createUpdateCompletion(completion_id: String, body: UpdateCompletionPostRequest) async throws -> ChatCompletionResponse? {
        let response = try await client.post(ApiPaths.aiPath("/chat/completions/\(completion_id)"), body: body)
        return response as? ChatCompletionResponse
    }

    /// Delete chat completion
    public func deleteCompletion(completion_id: String) async throws -> ChatCompletionDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/chat/completions/\(completion_id)"))
        return response as? ChatCompletionDeleteResponse
    }

    /// Update chat completion
    public func patchUpdateCompletion(completion_id: String, body: UpdateCompletionPatchRequest) async throws -> ChatCompletionResponse? {
        let response = try await client.patch(ApiPaths.aiPath("/v1/chat/completions/\(completion_id)"), body: body)
        return response as? ChatCompletionResponse
    }

    /// Update chat completion
    public func patchUpdateCompletionChat(completion_id: String, body: PatchUpdateCompletionRequest) async throws -> ChatCompletionResponse? {
        let response = try await client.patch(ApiPaths.aiPath("/chat/completions/\(completion_id)"), body: body)
        return response as? ChatCompletionResponse
    }

    /// Get chat completion
    public func getChatCompletion(completion_id: String) async throws -> ChatCompletionResponse? {
        let response = try await client.get(ApiPaths.aiPath("/management/chat/completions/\(completion_id)"))
        return response as? ChatCompletionResponse
    }

    /// Update chat completion
    public func updateChatCompletion(completion_id: String, body: UpdateChatCompletionRequest) async throws -> ChatCompletionResponse? {
        let response = try await client.post(ApiPaths.aiPath("/management/chat/completions/\(completion_id)"), body: body)
        return response as? ChatCompletionResponse
    }

    /// Delete chat completion
    public func deleteChatCompletion(completion_id: String) async throws -> ChatCompletionDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/management/chat/completions/\(completion_id)"))
        return response as? ChatCompletionDeleteResponse
    }

    /// List chat completions
    public func listCompletions(params: [String: Any]? = nil) async throws -> ChatCompletionList? {
        let response = try await client.get(ApiPaths.aiPath("/chat/completions"), params: params)
        return response as? ChatCompletionList
    }

    /// Create chat completion
    public func createChatCompletion(body: ChatCompletionRequest) async throws -> CreateChatCompletionResponse? {
        let response = try await client.post(ApiPaths.aiPath("/chat/completions"), body: body)
        return response as? CreateChatCompletionResponse
    }

    /// Get chat messages
    public func getChatMessages(completion_id: String, params: [String: Any]? = nil) async throws -> ChatMessageList? {
        let response = try await client.get(ApiPaths.aiPath("/management/chat/completions/\(completion_id)/messages"), params: params)
        return response as? ChatMessageList
    }

    /// List chat completions
    public func listChatCompletions(params: [String: Any]? = nil) async throws -> ChatCompletionList? {
        let response = try await client.get(ApiPaths.aiPath("/management/chat/completions"), params: params)
        return response as? ChatCompletionList
    }

    /// Get chat completion messages
    public func getMessages(completion_id: String, params: [String: Any]? = nil) async throws -> ChatMessageList? {
        let response = try await client.get(ApiPaths.aiPath("/chat/completions/\(completion_id)/messages"), params: params)
        return response as? ChatMessageList
    }
}
